torch/csrc/distributed/c10d/TCPStore.cpp

```diff
@@ -1049,8 +1053,15 @@ void TCPStore::waitForWorkers() {
       }
       const auto elapsed = std::chrono::duration_cast<std::chrono::seconds>(
           std::chrono::steady_clock::now() - start);
-      if (timeout_ != kNoTimeout && elapsed > timeout_) {
-        break;
+
+      if (timeout_ != kNoTimeout) {
+        TORCH_CHECK(
+            elapsed < timeout_,
```

Comment:
Nit: this isn't entirely the equivalent of the old check. It should be `elapsed <= timeout_`.

Reply:
This check will cause the server to fail, but clients will most likely get a connection reset error. Do we consider it the expected behavior?